src/main/scala/com/mwz/sonar/scala/scoverage/package.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+
+package object scoverage {
+
+  /** Used to mock the scoverage report parser in tests */
+  trait ScoverageReportParserAPI {
+    def parse(scoverageReportFilename: String, sourcesPrefix: String): ModuleCoverage
+  }
+
+  /**
+   *  The coverage information of an entire module.
+   *  It is composed of:
+   *    - the overall [[ScoverageMetrics]] of the module.
+   *    - the coverage information of each file of the module.
+   */
+  private[scoverage] final case class ModuleCoverage(
+    moduleScoverage: Scoverage,
+    filesCoverage: Map[String, FileCoverage]
+  )
+
+  /**
+   *  The coverage information of a file.
+   *  It is composed of:
+   *    - the overall [[ScoverageMetrics]] of the file.
+   *    - the coverage information of each line of the module.
+   */
+  private[scoverage] final case class FileCoverage(
+    fileScoverage: Scoverage,
+    linesCoverage: LinesCoverage
+  ) {
+
+    /** Merges two file coverages in one */
+    private[scoverage] def +(that: FileCoverage): FileCoverage = {
+      val mergedFileScoverage = this.fileScoverage + that.fileScoverage
+      val mergedLinesCoverage = this.linesCoverage ++ that.linesCoverage
+      FileCoverage(mergedFileScoverage, mergedLinesCoverage)
+    }
+  }
+
+  /**
+   *  The coverage information of the lines of a file.
+   *  Linenum -> Line hits
+   */
+  private[scoverage] type LinesCoverage = Map[Int, Int]
+
+  /** [[ScoverageMetrics]] of a component */
+  private[scoverage] final case class Scoverage(
+    totalStatements: Int,
+    coveredStatements: Int,
+    statementCoverage: Double,
+    branchCoverage: Double
+  ) {
+
+    /** Merges two scoverages metrics in one */
+    private[scoverage] def +(that: Scoverage): Scoverage = {
```

Comment:
The moment I saw this I thought this would be a good use case for a [`Semigroup`](https://typelevel.org/cats/typeclasses/semigroup.html) typeclass from cats to combine two Scoverage instances into one without reimplementing a custom `+` function, e.g. `scoverage1 |+| scoverage2`, but don't worry about it for now.
At some point though, perhaps when we decide to write a scapegoat module, I would like to start to make use of cats and cats-effect as there would be a lot of good use cases for those libraries in this project.

Reply:
Yeah me too, but I didn't wanted to add a new dependency only for one use.
But yeah, if we find more use cases i would be happy to use it.

Reply:
Pretty much a standard these days so I'm totally happy with importing it whenever we need it - IMHO the additional size of those dependencies shouldn't really be an issue.

Reply:
yeah, there is probably no need for adding it as a part of this PR